crates/router/src/core/payment_methods/transformers.rs

```diff
@@ -280,6 +332,72 @@ pub fn mk_add_card_response_hs(
     }
 }
 
+pub fn mk_add_card_response(
+    card: api::CardDetail,
+    response: AddCardResponse,
+    req: api::PaymentMethodCreate,
+    merchant_id: &str,
+) -> api::PaymentMethodResponse {
+    let mut card_number = card.card_number.peek().to_owned();
+    let card = api::CardDetailFromLocker {
+        scheme: None,
+        last4_digits: Some(card_number.split_off(card_number.len() - 4)),
+        issuer_country: None, // [#256] bin mapping
+        card_number: Some(card.card_number),
+        expiry_month: Some(card.card_exp_month),
+        expiry_year: Some(card.card_exp_year),
+        card_token: Some(response.external_id.into()), // [#256]
+        card_fingerprint: Some(response.card_fingerprint),
+        card_holder_name: None,
+    };
+    api::PaymentMethodResponse {
+        merchant_id: merchant_id.to_owned(),
+        customer_id: req.customer_id,
+        payment_method_id: response.card_id,
+        payment_method: req.payment_method,
+        payment_method_type: req.payment_method_type,
+        card: Some(card),
+        metadata: req.metadata,
+        created: Some(common_utils::date_time::now()),
+        recurring_enabled: false,           // [#256]
+        installment_payment_enabled: false, // #[#256]
+        payment_experience: Some(vec![api_models::enums::PaymentExperience::RedirectToUrl]), // [#256]
```

Comment:
What is the reason for hard-coding the `PaymentExperience` to `RedirectToUrl`?

Reply:
Add a comment in the type definition and use defaults instead. Or add a comment here itself.

Reply:
make it `None` , this field is basically the experience of different type of transaction for a particular PM which will be a `vec[]` with PMT with most successful txns rate as the first entry
In future if we are asked to choose through which PMT we should do the txns , we can refer this

Reply:
Sure, for right now though, I have set it to `None`